### PR TITLE
Third part of water uptake

### PR DESCRIPTION
**Pull request by pressel**
This PR brings in a port of two more functions from water_uptake. In particular: 

- modal_aero_water_uptake_rh_clearair
- modal_aero_wateruptake_wetaer

### CODE REVIEW

**Review by jeff-cohere — APPROVED** [review]
👍

File: src/mam4xx/water_uptake.hpp [hunk 1]
```diff
@@ -215,6 +215,91 @@ void modal_aero_kohler(const Real rdry_in, const Real hygro, const Real rh,
...
+    Real wetvol[AeroConfig::num_modes()], Real wtrvol[AeroConfig::num_modes()],
+    Real dgncur_awet[AeroConfig::num_modes()],
+    Real qaerwat[AeroConfig::num_modes()]) {
+
+  //-----------------------------------------------------------------------
+  // loop over all aerosol modes
+  for (int imode = 0; imode < AeroConfig::num_modes(); ++imode) {
+
+    const Real hystfac =
+        1.0 / haero::max(1.0e-5, (rhdeliques[imode] - rhcrystal[imode]));
```
Comment (odiazib) [inline src/mam4xx/water_uptake.hpp:244]: Bad constant?

File: src/mam4xx/water_uptake.hpp [hunk 2]
```diff
@@ -215,6 +215,91 @@ void modal_aero_kohler(const Real rdry_in, const Real hygro, const Real rh,
...
+
+  static constexpr Real rh_max = 0.98; // (BAD CONSTANT)
+  if (qs > h2ommr) {
+    rh = h2ommr / qs;
+  } else {
+    rh = rh_max;
+  }
+  rh = utils::min_max_bound(0.0, rh_max, rh);
+
+  static constexpr Real cldn_thresh = 1.0;
```
Comment (odiazib) [inline src/mam4xx/water_uptake.hpp:296]: Bad constant?

**Review by odiazib — APPROVED** [review]
It looks good.

File: src/mam4xx/water_uptake.hpp (hunk 1 above)
Comment (pressel) [reply src/mam4xx/water_uptake.hpp]: Done

File: src/mam4xx/water_uptake.hpp (hunk 2 above)
Comment (pressel) [reply src/mam4xx/water_uptake.hpp]: Done

**Pull request merged**